src/benchmark/operators/insert_constraint_benchmark.cpp

```diff
@@ -47,71 +44,107 @@ auto reset_table(bool use_constraints, int values_to_insert, int num_rows = 0) {
 
-    for (const auto& table_insert : table_inserts) {
-      auto table_context = TransactionManager::get().new_transaction_context();
-      table_insert->set_transaction_context(table_context);
-      table_insert->execute();
-      table_context->commit();
-    }
+    // Actually execute the operator and trigger the constraint satiesfied check
+    insert_op->execute();
+    table_context->commit();
   }
 }
 
-static void insert_ranges(benchmark::internal::Benchmark* b) {
-  for (uint32_t j = 1; j <= 5000; j *= 2) {
+static void insert_ranges_filled_tables(benchmark::internal::Benchmark* b) {
+  for (uint32_t j = 100; j <= 150000; j *= 2) {
+    // The bool determines if constraints are used
+    // The int determines how many rows are pre-inserted
     b->Args({true, j});
     b->Args({false, j});
   }
 }
 
-BENCHMARK_REGISTER_F(MicroBenchmarkBasicFixture, BM_InsertEmptyTableWithConstraint)->Apply(insert_ranges);
+// The iterations are limited to 100 since the table set up takes quiete long
+BENCHMARK_REGISTER_F(MicroBenchmarkBasicFixture, BM_InsertFilledTableWithConstraint)
+    ->Apply(insert_ranges_filled_tables)
+    ->Iterations(100);
+
+BENCHMARK_DEFINE_F(MicroBenchmarkBasicFixture, BM_InsertOnCompressedTable)(benchmark::State& state) {
+  prepare_prefilled_table(static_cast<int>(state.range(1)), state.range(0), true);
+  auto& manager = StorageManager::get();
+  // Retrieve table to keep track of the row count
+  auto table = manager.get_table("table");
+
+  // Pre define column structure since it can be
+  TableColumnDefinitions column_definitions;
+  column_definitions.emplace_back("column0", DataType::Int, true);
+  column_definitions.emplace_back("column1", DataType::Int, false);
 
-BENCHMARK_DEFINE_F(MicroBenchmarkBasicFixture, BM_InsertFilledTableWithConstraint)(benchmark::State& state) {
   while (state.KeepRunning()) {
+    // Pause Timing to set up test table
     state.PauseTiming();
 
-    auto table_inserts = reset_table(state.range(0), 1, static_cast<int>(state.range(1)));
+    // Since we don't call prepare_prefilled_table between the iterations we have to delete the temp table manually
+    if (manager.has_table("table_temp")) {
+      manager.drop_table("table_temp");
+    }
 
-    state.ResumeTiming();
+    // Create the operator to be measured. By adding values below 0 the uniqueness is guaranteed
+    auto table_temp = std::make_shared<Table>(column_definitions, TableType::Data, 1, UseMvcc::Yes);
+    manager.add_table("table_temp", table_temp);
+    const int row_count = static_cast<int>(table->row_count());
+    table_temp->append({-1 * row_count, -2 * row_count});
 
-    for (const auto& table_insert : table_inserts) {
-      auto table_context = TransactionManager::get().new_transaction_context();
-      table_insert->set_transaction_context(table_context);
-      table_insert->execute();
-      table_context->commit();
-    }
-  }
-}
+    auto gt = std::make_shared<GetTable>("table_temp");
```

Comment:
I think we decided against the `TableWrapper` because we didn't find a way to generate the MVCC data with the `TableGenerator`.
But you're right, that doesn't stop us from just putting our own table into the `TableWrapper`. I will try to setup the Benchmark with a fixture.

Reply:
Didn't mention it here yet, but the constraint benchmark setup was switched to the `TableWrapper` in 1963d223f6488d002fcabbb2c003a0aad7ac23fe